controllers/application_controller.go

```diff
@@ -96,7 +96,7 @@ func (r *ApplicationReconciler) getNewApplicationStatus(ctx context.Context, app
 		Objects: objectStatuses,
 	}
 	newApplicationStatus.ComponentsReady = fmt.Sprintf("%d/%d", countReady, len(objectStatuses))
-	if aggReady && errs != nil {
+	if aggReady && errs == nil {
 		setReadyCondition(newApplicationStatus, "ComponentsReady", "all components ready")
 	} else {
 		setNotReadyCondition(newApplicationStatus, "ComponentsNotReady", "some components not ready")
```

Comment:
is it possible that all components are ready but there's an error ?
I assume yes, the message / status should reflect that or it's still confusing. 

The status would be in error with "some components not ready", as a user I would check which components isn't ready but won't find any.... 
Could we add a separate case:
something like :
```
if errs != nil { 
  "there's an error"
} else if aggReady {
 "all components ready"
} else 
{
 "some components not ready" // 'some' could be replaced with (len(objectStatuses)-countReady)
}
```

Reply:
There are two types of conditions: `Ready` and `Error` and we need to set both.
I think we can leverage the `Unknown` status for the `Ready` condition:
```
if errs != nil {
   //set ready condition to unknown
} else if aggReady {
  "all components ready"
} else {
    "x components not ready"
}

if errs != nil {
   setErrorCondition
} else {
   clearErrorCondition
}
```